packages/frontend-2/components/common/EditableTitleDescription.vue

```diff
@@ -0,0 +1,194 @@
+        v-if="canEdit"
+        class="shrink-0 ml-2 mt-3 w-4 h-4 opacity-0 group-hover:opacity-100 transition text-foreground-2"
+      />
+    </div>
+
+    <!-- Editable Description -->
+    <div class="mt-3 flex gap-x-2 group">
+      <div class="hidden md:inline-block shrink-0 text-foreground-2 mt-0.5">
+        <InformationCircleIcon class="w-5 h-5" />
+      </div>
+      <label>
+        <div class="sr-only">Edit description</div>
+        <div
+          class="grow-textarea"
+          :data-replicated-value="description"
+          :class="descriptionInputClasses"
+        >
+          <textarea
+            v-model="description"
+            :class="[
+              ...descriptionInputClasses,
+              description ? 'focus:min-w-0' : 'min-w-[260px]'
+            ]"
+            :placeholder="description ? undefined : 'Click here to add a description.'"
+            :disabled="isDisabled"
+            rows="1"
+            spellcheck="false"
+            maxlength="1000"
+            :cols="
+              description && description?.length < 20 ? description.length : undefined
+            "
+            @keydown="onInputKeydown"
+            @blur="onBlur('description')"
+            @input="onDescriptionChange"
+          />
+        </div>
+      </label>
+      <div class="shrink-0 ml-2 mt-1 text-foreground-2">
+        <PencilIcon
+          v-if="canEdit"
+          class="w-4 h-4 opacity-0 group-hover:opacity-100 transition text-foreground-2"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { PencilIcon } from '@heroicons/vue/20/solid'
+import { InformationCircleIcon } from '@heroicons/vue/24/outline'
+import { debounce } from 'lodash-es'
+
+const props = defineProps({
+  initialTitle: String,
+  initialDescription: String,
+  canEdit: Boolean,
+  isDisabled: Boolean
+})
+
+const emit = defineEmits(['update:title', 'update:description'])
+
+const title = ref(props.initialTitle)
+const description = ref(props.initialDescription)
+
+const lastTitleValue = ref(props.initialTitle)
+const lastDescriptionValue = ref(props.initialDescription)
+
+const emitTitle = () => emit('update:title', title.value)
+const emitDescription = () => emit('update:description', description.value)
+
+const debouncedEmitTitle = debounce(emitTitle, 2000)
+const debouncedEmitDescription = debounce(emitDescription, 2000)
+
+const titleInputClasses = computed(() => [
+  'h3 tracking-tight border-0 border-b-2 transition focus:border-outline-3 max-w-full',
+  'p-0 bg-transparent border-transparent focus:outline-none focus:ring-0'
+])
+
+const descriptionInputClasses = computed(() => [
+  'normal placeholder:text-foreground-2 text-foreground-2 focus:text-foreground',
+  'border-0 border-b-2 focus:border-outline-3',
+  'p-0 bg-transparent border-transparent focus:outline-none focus:ring-0'
+])
+
+const onInputKeydown = (e: KeyboardEvent) => {
+  if (e.code === 'Enter' && e.target instanceof HTMLElement) {
```

Comment:
We only want Enter to submit for 'title'. 'description' allows multiple lines and thus should not submit on enter.